### PR TITLE
[d3d9] Add extended swapchain interfaces

### PR DESCRIPTION
**Pull request by Joshua-Ashton**

### CODE REVIEW

File: src/d3d9/d3d9_swapchain.cpp [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 #include "d3d9_hud.h"
 #include "d3d9_window.h"
+#include <vulkan/vulkan_core.h>
```
Comment (pchome) [inline src/d3d9/d3d9_swapchain.cpp:7]: ...

Comment (K0bin) [reply src/d3d9/d3d9_swapchain.cpp:7]: VSCode strikes again.

**Comment by SirSAC** [comment]
Is not a good idea to add to "d3d11_swapchain.cpp" too?

**Comment by K0bin** [comment]
@SirSAC D3D11 already supports all of this.

**Comment by SirSAC** [comment]
These fixed the super sharp edges of shadow of the torches on the night on Heaven Benchmark, without these changes the shadow looks anime instead of real life.

**Comment by K0bin** [comment]
@SirSAC That's completely impossible. Not a single game uses this code and even if it did, it has nothing to do with what you're describing.

**Comment by SirSAC** [comment]
> @SirSAC That's completely impossible. Not a single game uses this code and even if it did, it has nothing to do with what you're describing.

maybe it depends by hardware i have, it just did.

**Comment by K0bin** [comment]
> maybe it depends by hardware i have, it just did.

No. Code that is never executed cannot make any difference. Just placebo.

**Comment by SirSAC** [comment]
> > maybe it depends by hardware i have, it just did.
> 
> No. Code that is never executed cannot make any difference. Just placebo.

what is placebo? is some kind of imagination or illusion?
english is not my mother tongue

**Comment by Riesi** [comment]
https://ro.wikipedia.org/wiki/Placebo

**Comment by SirSAC** [comment]
> https://ro.wikipedia.org/wiki/Placebo

how did u know im romanian?

**Comment by Blisto91** [comment]
It's tagged on your GitHub profile. Anyway enough communing about that.

**Comment by SirSAC** [comment]
> It's tagged on your GitHub profile. Anyway enough commenting about that so the PR comments doesn't get too far off track.

PR?

**Comment by Blisto91** [comment]
Short for Pull Request. When someone want's you to pull your changes into your git project (like this one here). On Gitlab the same thing is called a MR which is short for Merge Request.

**Comment by SirSAC** [comment]
> Short for Pull Request. When someone want's you to pull their changes into your git project (like this one here). On Gitlab the same thing is called a MR which is short for Merge Request.

ok, thanks for explaining the terms

**Pull request merged**